examples/v2/Vector/VecIntAndReal.F90

```diff
@@ -0,0 +1,86 @@
+! This is an example for integer and real vector. 
+! It can be changed to any intrinsic type easliy.
+! 1) create a module for each type
+! 2) use alias when using different  vector types
+
+module IntegerVec_mod
+#define T integer
+#define Vector VectorInt
+#define VectorIterator VectorIntIterator
+#include "vector/template.inc"
+#undef T
+#undef Vector
+#undef VectorIterator
+end module IntegerVec_mod
+
+module Real64Vec_mod
+use, intrinsic :: iso_fortran_env
+
+#define T real
+#define T_KINDLEN(context) (kind=real64)
+#include "vector/template.inc"
```

Comment:
I am torn here.    I intentionally created shortcuts for the variant kinds of intrinsics:

```
#define T __REAL32
...
```
OTOH, it is useful to show others how one would add parameters to the spec for non-intrinsic types.

Reply:
I think I can use the shortcut as `T_KINDLEN(context)` is also illustrated in the module below that uses `iso_fortran_env`

Reply:
Done [20f61b0](https://github.com/Goddard-Fortran-Ecosystem/gFTL/pull/239/commits/20f61b09a1fa7210bcdb671b254216a5ab6d7a8f)

Reply:
Should the compilation of these files be added to the tests?

Reply:
I had removed the `use, intrinsic :: iso_fortran_env`. I didn't realise that `__REAL64` was using this.
[07b41c4](https://github.com/Goddard-Fortran-Ecosystem/gFTL/pull/239/commits/07b41c4cf00e7e95f3d02c21751b28eabd8d7986)
It is now compiling correctly again

Reply:
OK - should not be be difficult after all.

I'll open an issue and sketch the steps.